examples/threading/threading.c

```diff
@@ -13,7 +14,22 @@ void* threadfunc(void* thread_param)
 
     // TODO: wait, obtain mutex, wait, release mutex as described by thread_data structure
     // hint: use a cast like the one below to obtain thread arguments from your parameter
-    //struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+    struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+    usleep(thread_func_args->wait_to_obtain_ms);
+    int lock = pthread_mutex_lock(thread_func_args->mutex);
+    if(lock != 0){
+    	thread_func_args->thread_complete_success = false;
+    	printf("\r\nMutex Lock Failed");
+    }
+    else{
+    	usleep(thread_func_args->wait_to_release_ms);
```

Comment:
@srki3050 can you be sure usleep completed the wait here without checking return status?